src/views/virtualmachines/list/VirtualMachinesList.tsx

```diff
@@ -152,6 +156,8 @@ const VirtualMachinesList: FC<VirtualMachinesListProps> = ({ kind, namespace })
 
   const noVMs = isEmpty(unfilteredData) && !vmsFilteredWithProxy;
 
+  const allVMsSelected = data?.length === selectedVMs.length;
```

Comment:
that will trigger a lot of rerenders, did u checked? compared with useMemo and without?

Reply:
I can use useMemo, yeah but I would suggest investigating deeply as in this component there are many things that trigger re-render. Starting from rowData and columnLayout. Those components get created on the fly at every rendering so they alone trigger a children re-rendering